reviewing/dependencies-hierarchy/test/getTree.test.ts

```diff
@@ -147,4 +147,334 @@ describe('getTree', () => {
+              alias: 'inflight',
+              dependencies: expect.arrayContaining([
+                // depth 2
+                expect.objectContaining({
+                  // The "once" package is first seen here at depth 2.
+                  alias: 'once',
+                }),
+              ]),
+            }),
+
+            // depth 1
+            expect.objectContaining({
+              alias: 'once',
+              dependencies: [
+                // The "once" package is seen again at depth 1. The "once"
+                // package contains a "wrappy" package that should be listed.
+                expect.objectContaining({ alias: 'wrappy' }),
+              ],
+            }),
+          ]),
+        }),
+      ])
+    })
+
+    test('revisiting package at higher depth does not print extra dependenices', () => {
+      // This tests the "glob" npm package on a subset of its dependency tree.
+      // Requested depth (max depth - current depth) shown in square brackets.
+      //
+      // root
+      // └─┬ a [2]
+      //   ├─┬ b [1]   <-- 1st time "b" is seen.
+      //   │ └── c [0]
+      //   └─┬ d [1]
+      //     └── b [0] <-- 2nd time "b" is seen. Dependencies should not be printed since "max depth === current depth".
+      const version = '1.0.0'
+      const currentPackages = generateMockCurrentPackages(version, {
+        root: ['a'],
+        a: ['b', 'd'],
+        b: ['c'],
+        d: ['b'],
+      })
+      const rootDepPath = refToRelativeOrThrow(version, 'root')
+
+      const result = getTree({
+        ...commonMockGetTreeArgs,
+        currentDepth: 0,
+        maxDepth: 2,
+        currentPackages,
+        wantedPackages: currentPackages,
+      }, [rootDepPath], rootDepPath)
+
+      expect(normalizePackageNodeForTesting(result)).toEqual([
+        expect.objectContaining({
+          alias: 'a',
+          dependencies: [
+
+            expect.objectContaining({
+              alias: 'b',
+              dependencies: [
+                expect.objectContaining({
+                  alias: 'c',
+                }),
+              ],
+            }),
+
+            expect.objectContaining({
+              alias: 'd',
+              dependencies: [
+                expect.objectContaining({
+                  alias: 'b',
+
+                  // The "b" package has a "c" dependency, but it should not be
+                  // printed since the max depth was reached.
+                  dependencies: undefined,
+                }),
+              ],
+            }),
+          ],
+        }),
+      ])
+    })
+  })
+
+  // This group of tests attempts to check that situations when the "fully
+  // visited cache" can be reused is correct.
+  //
+  // This doesn't test the cache directly, but sets up situations that would
+  // result in incorrect output if the cache was used when it's not supposed to.
+  describe('fully visited cache optimization handles requested depth correctly', () => {
```

Comment:
This group of tests was also added in the latest iteration of the PR. It makes me much more confident the "fully visited cache" is correct and is less likely to break in the future as others edit this package.